### ISSUE TITLE
bug(overlay): cdk-overlay-container a11y failure (not contained by landmark)

### ISSUE THREAD

**Issue by pburkindine** [angular/components#20001]
#### Reproduction

Steps to reproduce:
1. Visit https://overlay-cont-a11y.stackblitz.io/
2. Open the mat-select so a `cdk-overlay-container` is present
2. Run axe plugin to check 508 (https://www.deque.com/axe/browser-extensions/)

axe gives the following error regarding the `cdk-overlay-container`, since it is outside the `main` landmark:

```
All page content must be contained by landmarks
```

![Screen Shot 2020-07-15 at 8 46 27 PM](https://user-images.githubusercontent.com/20215564/87617138-63f00300-c6dc-11ea-829d-8420dd8ad013.png)

 
 
#### Expected Behavior

Overlay container should not throw an a11y violation

#### Actual Behavior

See above

#### Environment

- Angular: 10.0.3
- CDK/Material: 10.0.1
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu):  OSX

**Comment by ianoxwell** [comment]
Unfortunately this is still an issue in Angular 12.1.1

Current work around is in app.component.ts
```
import { OverlayContainer } from '@angular/cdk/overlay';
...

constructor(private overlayContainer: OverlayContainer) { }

ngOnInit(): void {
   this.overlayContainer.getContainerElement().setAttribute('role', 'region');
}
```

This applies the role of 'region' to the overlay
[https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Region_role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Region_role)

There may be a more appropriate role?

**Comment by crisbeto** [comment]
I don't think that any of the landmark roles would be appropriate for the overlay container since it can contain all sorts of overlays. From the `region` role specifically:
> The region role should be reserved for sections of content sufficiently important that users will likely want to navigate to the section easily **and to have it listed in a summary of the page**.

It's worth noting that the lack of a role doesn't have an impact on the user's experience, it only gets flagged by automated tools that don't have any context to work with. I'm closing the issue, because I don't see an appropriate role that can be applied and the workaround above can be used if you want to get rid of the tooling warnings.
